File: sqlparser/src/sql/SqlOrderSpec.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

public class SqlOrderSpec implements ISqlCloneable {
  private /*@*/final/*@*/ /*#readonly#*/ SqlExpression column;
  private /*@*/final/*@*/ /*#readonly#*/ SortOrder order;
  private /*@*/final/*@*/ /*#readonly#*/ boolean isNullsFirst;
  private /*@*/final/*@*/ /*#readonly#*/ boolean hasNulls;

  public SqlOrderSpec(SqlExpression col, SortOrder order, boolean isNullsFirst, boolean hasNulls) {
    this.column = col;
    this.order = order;
    this.isNullsFirst = isNullsFirst;
    this.hasNulls = hasNulls;
  }

  public String getResolvedColumnName() {
    if (column instanceof SqlColumn) {
      return ((SqlColumn) column).getColumnName();
    } else if (column instanceof SqlValueExpression) {
      return ((SqlValueExpression) column).evaluate().getValueAsString("");
    }
    return null;
  }

  public SortOrder getOrder() {
    return order;
  }

  public SqlExpression getExpr() {
    return this.column;
  }

  public boolean isNullsFirst() {
    return this.isNullsFirst;
  }

  public boolean hasNulls() {
    return this.hasNulls;
  }

  public Object clone() {
    return new SqlOrderSpec(this.column, this.order, this.isNullsFirst, this.hasNulls);
  }
}

File: sqlparser/src/sql/csgen/SqlOrderSpec.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public class SqlOrderSpec : ISqlCloneable {
  private  readonly SqlExpression column;
  private  readonly SortOrder order;
  private  readonly bool isNullsFirst;
  private  readonly bool hasNulls;

  public SqlOrderSpec(SqlExpression col, SortOrder order, bool isNullsFirst, bool hasNulls) {
    this.column = col;
    this.order = order;
    this.isNullsFirst = isNullsFirst;
    this.hasNulls = hasNulls;
  }

  public string GetResolvedColumnName() {
    if (column is SqlColumn) {
      return ((SqlColumn) column).GetColumnName();
    } else if (column is SqlValueExpression) {
      return ((SqlValueExpression) column).Evaluate().GetValueAsString("");
    }
    return null;
  }

  public SortOrder GetOrder() {
    return order;
  }

  public SqlExpression GetExpr() {
    return this.column;
  }

  public bool IsNullsFirst() {
    return this.isNullsFirst;
  }

  public bool HasNulls() {
    return this.hasNulls;
  }

  public Object Clone() {
    return new SqlOrderSpec(this.column, this.order, this.isNullsFirst, this.hasNulls);
  }
}
}

